Umee-Bots/Large-Treasury-Token-Price-Change/src/agent.spec.ts

```diff
@@ -0,0 +1,396 @@
+      ethers.BigNumber.from(calls.latestTimestamp),
+      ethers.BigNumber.from(calls.latestTimestamp),
+      ethers.BigNumber.from(0),
+    ],
+  });
+}
+
+describe("large treasury token price change bot", () => {
+  let mockProvider: MockEthersProvider = new MockEthersProvider();
+  let provider: ethers.providers.Provider = mockProvider as any as ethers.providers.Provider;
+  let initialize: Initialize;
+  let handleBlock: HandleBlock;
+
+  beforeEach(() => {
+    mockProvider = new MockEthersProvider();
+    provider = mockProvider as any as ethers.providers.Provider;
+  });
+
+  describe("initialize", () => {
+    it("should load empty monitoring array if config.tokens is empty", async () => {
+      initialize = provideInitialize(provider, { tokens: [] });
+
+      await initialize();
+
+      expect(agent.getTokens()).toStrictEqual([]);
+    });
+
+    it("should throw an error if the last feed answer is negative", async () => {
+      const feedAddress = createAddress("0xfeed");
+
+      mockFeedCalls(mockProvider, feedAddress, {
+        description: "description",
+        decimals: 18,
+        latestAnswer: "-1",
+        latestTimestamp: "0",
+      });
+
+      initialize = provideInitialize(provider, {
+        tokens: [{ chainlinkFeedAddress: feedAddress, intervalSeconds: "3600" }],
+      });
+
+      await expect(initialize).rejects.toThrow("Negative feed answer");
+      expect(agent.getTokens()).toStrictEqual([]);
+    });
+
+    it("should successfully load monitoring data using the agent config", async () => {
+      const feedAddress = createAddress("0xfeed");
+      const feedCalls = {
+        description: "description",
+        decimals: 18,
+        latestAnswer: "0",
+        latestTimestamp: "0",
+      };
+      const agentConfig = {
+        tokens: [
+          {
+            chainlinkFeedAddress: feedAddress,
+            intervalSeconds: "3600",
+            absoluteThreshold: "0",
+            percentageThreshold: "0",
+          },
+        ],
+      };
+
+      mockFeedCalls(mockProvider, feedAddress, feedCalls);
+
+      initialize = provideInitialize(provider, agentConfig);
+      await initialize();
+
+      const tokens = agent.getTokens();
+
+      expect(tokens[0].description).toStrictEqual(feedCalls.description);
+      expect(tokens[0].decimals).toStrictEqual(feedCalls.decimals);
+      expect(tokens[0].lastTimestamp.toString()).toStrictEqual(feedCalls.latestTimestamp);
+      expect(tokens[0].lastAnswer.toString(10)).toStrictEqual(feedCalls.latestAnswer);
+      expect(tokens[0].absoluteThreshold?.toString(10)).toStrictEqual(agentConfig.tokens[0].absoluteThreshold);
+      expect(tokens[0].ratioThreshold?.shiftedBy(2).toString(10)).toStrictEqual(
+        agentConfig.tokens[0].percentageThreshold
+      );
+      expect(tokens[0].intervalSeconds.toString()).toStrictEqual(agentConfig.tokens[0].intervalSeconds);
+    });
+  });
+
+  describe("handleBlock", () => {
```

Comment:
```
    beforeAll(() => {
      handleBlock = provideHandleBlock();
    });
```
can be added inside describe, in order to avoid repetition of calling `handleBlock = provideHandleBlock();` in each case.

Reply:
Done!